extensions/sql-migration/src/api/azure.ts

```diff
@@ -159,6 +160,22 @@ export async function createSqlMigrationService(account: azdata.Account, subscri
 	if (response.errors.length > 0) {
 		throw new Error(response.errors.toString());
 	}
+	const asyncUrl = response.response.headers['azure-asyncoperation'];
+	const maxRetry = 5;
+	let i = 0;
+	for (i = 0; i < maxRetry; i++) {
+		const asyncResponse = await api.makeAzureRestRequest(account, subscription, asyncUrl.replace('https://management.azure.com/', ''), azurecore.HttpRequestMethod.GET, undefined, true);
+		const creationStatus = asyncResponse.response.data.status;
+		if (creationStatus === 'Succeeded') {
+			break;
+		} else if (creationStatus === 'Failed') {
+			throw new Error(asyncResponse.errors.toString());
+		}
+		await new Promise(resolve => setTimeout(resolve, 3000)); //adding  3 sec delay before getting creation status
+	}
+	if (i === 6) {
```

Comment:
Generally the pattern I see used is 
```
while (true) {
   if (i === maxRetry) { throw Error(...); }
   ...
}
```

But at the very least you should do 

`if (i === maxRetry)`

(because right now the logic isn't even correct I believe, it should never be getting to 6. And hardcoding numbers like that isn't great, makes it riskier if maxRetry is changed in the future)